providers/azure/azure_test.go

```diff
@@ -132,6 +132,16 @@ container: "MyContainer"`),
 		wantFailParse:    false,
 		wantFailValidate: false,
 	},
+	{
+		name: "Valid User Assigned and Connection String set",
+		config: []byte(`storage_account: "myAccount"
+storage_account_key: ""
```

Comment:
how is this valid  since storage_account_key and storage_connection_string cannot both be set?